### PR TITLE
feat: add supply and borrower count to core pool

### PR DESCRIPTION
**Pull request by coreyar**
Adds supply and borrower count to core subgraph
A borrowerCountAdjusted value is added that removes accounts that were liquidated with only dust left

### CODE REVIEW

File: subgraphs/isolated-pools/schema.graphql [hunk 1]
```diff
@@ -184,11 +184,11 @@ type AccountVToken @entity {
     enteredMarket: Boolean!
 
     "Total amount of underling redeemed"
```
Comment (gleiser-oliveira) [inline subgraphs/isolated-pools/schema.graphql:186]:
```suggestion
    "Total amount of underlying redeemed"
```

File: subgraphs/venus/schema.graphql [hunk 2]
```diff
@@ -67,6 +67,15 @@ type Market @entity {
     totalXvsDistributedMantissa: BigInt!
     "vToken decimal length"
     vTokenDecimals: Int!
+
+    "Number of accounts currently supplying to this market"
+    supplierCount: BigInt!
+
+    "Number of accounts with reasonable borrow positions in this market (excludes accounts with dust (potentially left over from liquidations))"
```
Comment (gleiser-oliveira) [inline subgraphs/venus/schema.graphql:74]:
```suggestion
    "Number of accounts with reasonable borrow positions in this market (excludes accounts with dust potentially left over from liquidations)"
```

File: subgraphs/venus/src/constants/index.ts [hunk 3]
```diff
@@ -20,13 +20,13 @@ export const zeroBigInt32 = BigInt.fromString('0');
...
-  EXIT_MARKET,
+  'MINT',
+  'REDEEM',
+  'BORROW',
+  'REPAY',
+  'SEIZE',
+  'LIQUIDATE',
+  'TRANSFER',
+  'ENTER_MARKET',
+  'EXIT_MARKET',
```
Comment (gleiser-oliveira) [inline subgraphs/venus/src/constants/index.ts:31]: From the outside it looks the same as before, just that it's not using the constants declared above

File: subgraphs/venus/src/mappings/vToken.ts [hunk 4]
```diff
@@ -150,42 +144,34 @@ export const handleBorrow = (event: Borrow): void => {
...
   borrow.borrower = event.params.borrower;
   borrow.blockNumber = event.block.number.toI32();
   borrow.blockTime = event.block.timestamp.toI32();
   borrow.underlyingSymbol = market.underlyingSymbol;
   borrow.save();
+
+  if (event.params.accountBorrows == event.params.borrowAmount) {
+    // if both the accountBorrows and the borrowAmount are the same, it means the account is a new borrower
+    market.borrowerCount = market.borrowerCount.plus(oneBigInt);
+    market.borrowerCountAdjusted = market.borrowerCountAdjusted.plus(oneBigInt);
```
Comment (gleiser-oliveira) [inline subgraphs/venus/src/mappings/vToken.ts:172]: Question: does it make sense to apply the same dust rule here if you borrow a real small amount? Although I do see the difference between borrowing and liquidating

File: subgraphs/venus/src/mappings/vToken.ts [hunk 5]
```diff
@@ -225,43 +211,39 @@ export const handleRepayBorrow = (event: RepayBorrow): void => {
...
   repay.blockTime = event.block.timestamp.toI32();
   repay.underlyingSymbol = market.underlyingSymbol;
   repay.payer = event.params.payer;
   repay.save();
+
+  if (event.params.accountBorrows.equals(zeroBigInt32)) {
+    market.borrowerCount = market.borrowerCount.minus(oneBigInt);
+    market.borrowerCountAdjusted = market.borrowerCountAdjusted.minus(oneBigInt);
+    market.save();
+  } else if (event.params.accountBorrows.le(new BigInt(10))) {
```
Comment (gleiser-oliveira) [inline subgraphs/venus/src/mappings/vToken.ts:241]: nit: I'd declare this 10 a constant (DUST_THRESHOLD maybe?) and then use it here

File: subgraphs/venus/tests/VToken/index.test.ts [hunk 6]
```diff
@@ -268,297 +278,344 @@ describe('VToken', () => {
...
-        halfBorrowAmountTokens,
-        zeroBigInt32,
-        zeroBigInt32,
-      );
-
-      handleRepayBorrow(repayEvent);
-    });
+    handleRepayBorrow(repayEvent);
+    assert.fieldEquals('Market', aaaTokenAddress.toHex(), 'borrowerCount', '1');
+    assert.fieldEquals('Market', aaaTokenAddress.toHex(), 'borrowerCountAdjusted', '1');
```
Comment (gleiser-oliveira) [inline subgraphs/venus/tests/VToken/index.test.ts:619]: I'd also test out this guy based on the dust threshold

Comment (coreyar) [reply subgraphs/venus/tests/VToken/index.test.ts:619]: That is what this test is doing but it isn't working correctly as you point out

**Review by gleiser-oliveira — APPROVED** [review]

**Pull request merged**